lua/rocks/api.lua

```diff
@@ -99,16 +97,36 @@ function api.get_rocks_toml_path()
     return config.config_path
 end
 
+---@class RockSpec: { name: rock_name, version?: string, opt?: boolean, [string]: unknown }
+---@brief [[
+---        { name: rock_name, version?: string, opt?: boolean, [string]: V }
+---
+---Specification for a rock in rocks.toml.
+---@brief ]]
+
+---@class RocksToml: { rocks?: RockSpec[], plugins?: RockSpec[], [string]: unknown }
+---@brief [[
+---        { rocks?: RockSpec[], plugins?: RockSpec[], [string]: V }
+---
+---Content of rocks.toml
+---@brief ]]
+
 ---Returns a table with the parsed rocks.toml file.
 ---If the file doesn't exist a file with the default configuration will be created.
----@return table rocks_toml_file
+---@return RocksToml
 function api.get_rocks_toml()
-    local config_file = fs.read_or_create(config.config_path, constants.DEFAULT_CONFIG)
-    return require("toml").decode(config_file)
+    return config.get_rocks_toml()
+end
+
+---Returns a table with the rock specifications parsed from the rocks.toml file.
+---If the file doesn't exist a file with the default configuration will be created.
+---@return RockSpec[]
+function api.get_user_rocks()
+    return config.get_user_rocks()
 end
```

Comment:
Note: I have moved the implementations to `config.internal` to prevent cyclic dependencies.